### DISCUSSION TITLE
Usage of tor.Dial for .onion addresses

### DISCUSSION THREAD

**Discussion by hieblmi** [lightningnetwork/lnd#6019] [Q&A]
Hi team,

I am working on my first contribution to lnd.
In that context I would like to know if it is correct to use the `lnd/tor/tor.go` package to dial .onion addresses.
Would the following in principle return a valid tor dialer?
`tor.Dial("example.onion", "localhost:9050", false, false, tor.DefaultConnTimeout)` where the `localhost` is running a tor proxy on port `9050`.

Let me know if you need more context of my use-case.

Thanks for your help!

**Comment by guggero** [comment]
That sounds about right, yes. Though that will return a `net.Conn`. If you need a `net.Dialer`, you probably need to wrap things a bit. Also take a look at `tor.ProxyNet`.

**Reply by hieblmi** [reply]
Thanks for that, I made some progress and would like to keep this discussion open for further questions :-)
